lib/api/metadataSearch.js

```diff
@@ -0,0 +1,410 @@
+const querystring = require('querystring');
+const { errors, versioning, s3middleware } = require('arsenal');
+const constants = require('../../constants');
+const services = require('../services');
+const { metadataValidateBucket } = require('../metadata/metadataUtils');
+const collectCorsHeaders = require('../utilities/collectCorsHeaders');
+const escapeForXml = s3middleware.escapeForXml;
+const { pushMetric } = require('../utapi/utilities');
+const validateSearchParams = require('../api/apiUtils/bucket/validateSearch');
+const parseWhere = require('../api/apiUtils/bucket/parseWhere');
+const versionIdUtils = versioning.VersionID;
+const monitoring = require('../utilities/monitoringHandler');
+const { generateToken, decryptToken }
+    = require('../api/apiUtils/object/continueToken');
+const { config } = require('../Config');
+
+// do not url encode the continuation tokens
+const skipUrlEncoding = new Set([
+    'ContinuationToken',
+    'NextContinuationToken',
+]);
+
+/* Sample XML response for GET bucket objects V2:
```

Comment:
Do we need to keep these comments? Looks like they are equal. or maybe we can document here the metadata-related calls and expected responses?

Reply:
solved https://github.com/scality/cloudserver/pull/4224/commits/f91cb730fb14781a5a3786de59750ffe0cb79251